onnxruntime/core/codegen/common/common.cc

```diff
@@ -140,18 +140,20 @@ std::unique_ptr<ComputeCapability> ToCapacity(const onnxruntime::GraphViewer& gr
 
   for (const auto& node_index : subgraph->nodes) {
     const auto& node = *graph.GetNode(node_index);
+    auto process_input_fn = 
+      [&meta_def, &node, &node_indices](const onnxruntime::NodeArg& def, size_t) {
+        const onnxruntime::Node* input_node = GetInputNode(node, &def);
+        bool input_from_subgraph = (input_node && node_indices.count(input_node->Index()));
+        if (!input_from_subgraph) {
+          // input is from weights or outside of graph
+          meta_def->inputs.push_back(def.Name());
+        }
+        return Status::OK();
+      };
     // handle current graph's inputs
-    node.ForEachWithIndex(
-        node.InputDefs(),
-        [&meta_def, &node, &node_indices](const onnxruntime::NodeArg& def, size_t) {
-          const onnxruntime::Node* input_node = GetInputNode(node, &def);
-          bool input_from_subgraph = (input_node && node_indices.count(input_node->Index()));
-          if (!input_from_subgraph) {
-            // input is from weights or outside of graph
-            meta_def->inputs.push_back(def.Name());
-          }
-          return Status::OK();
-        });
+    node.ForEachWithIndex(node.InputDefs(), process_input_fn);
+    // nodes' implicit inputs also need to be collected.
+    node.ForEachWithIndex(node.ImplicitInputDefs(), process_input_fn);
```

Comment:
nit: may be helpful to say the implicit inputs need to be promoted to being explicit inputs for everything to work